### PR TITLE
[DLS-9459] Amending cancellation submission time

### PR DESCRIPTION
**Pull request by JakeReid2020**

### CODE REVIEW

**Review by GinnyThomas — CHANGES_REQUESTED** [review]
Is this supposed to be the entire ticket or just 1 of the 4 variations and the others are to follow?  If the latter, please remove the additional controllers out of the ticket.  If the former, please complete the work - updateDone/Sent, etc. controllers for other journeys, except cancellation, are not completed. View pages are not testing for times, which is probably why they were incorrect and we didn't previously notice.    Also, ticket indicates: "Also make sure that a new instance or empty user answers are being generated if. user were to start a new variation journey - i.e we do not want submittedOn to have a stamp from a previous journey. "  This is not currently the case - UAs loading onto Select Change page shows prior journey info.  You might want to check with Christine to see where she wants UAs cleared

File: app/controllers/cancelRegistration/CancellationRequestDoneController.scala [hunk 1]
```diff
@@ -37,40 +39,47 @@ class CancellationRequestDoneController @Inject()(
...
+            ))
+          )
+
+          Ok(view(
+            formattedDate, formattedTime,
+            returnPeriodStart, returnPeriodEnd,
+            deadlineStart, deadlineEnd,
+            request.subscription.orgName, Seq(cancelRegistrationSummary)))
+
+        case None => genericLogger.logger.error(s"[SoftDrinksIndustryLevyService [submitVariation] - unexpected response while attempting to retreive userAnswers submittedOnDate")
```
Comment (GinnyThomas) [inline app/controllers/cancelRegistration/CancellationRequestDoneController.scala:81]: The None on SubmittedOn means it is empty.  Should there be an Internal server error here or should the user go back to CYA?

Comment (GinnyThomas) [reply app/controllers/cancelRegistration/CancellationRequestDoneController.scala:81]: This question would also be the same for the other 3 done/sent controllers

**Review by GinnyThomas — CHANGES_REQUESTED** [review]
Original PR review statement/questions unaddressed

File: app/controllers/cancelRegistration/CancellationRequestDoneController.scala (hunk 1 above)
Comment (JakeReid2020) [reply app/controllers/cancelRegistration/CancellationRequestDoneController.scala:81]: Underneath is the internal server error for this case

**Review by GinnyThomas — CHANGES_REQUESTED** [review]
waiting for ALL PR review questions/comments to be adequately addressed. 2nd reviewer may need to be requested

**Comment by Chrissystk** [comment]
Essentially just the confirmation pages need to route to the beginning of the journey if there is no submittedOn time not return the internal server error. In terms of the clearing the submittedOn time from userAnswers it is already being done in the SelectChangeOrchestrator so nothing needs to be done there.

**Review by Chrissystk — DISMISSED** [review]

**Review by Chrissystk — DISMISSED** [review]

**Review by GinnyThomas — APPROVED** [review]

**Pull request merged**